Python/pywarpx/WarpX.py

```diff
@@ -101,7 +103,16 @@ def write_inputs(self, filename='inputs', **kw):
 
         with open(filename, 'w') as ff:
 
+            prefix_old = ''
             for arg in argv:
+                # This prints the name of the input group (prefix) as a header
+                # before each group to make the input file more human readable
+                prefix_new = re.split(' |\.', arg)[0]
```

Comment:
Just a note: this usage is slow and rebuilds the regex in the `for` loop, although it is constant. If you re-use the same regex multiple times, better pre-compile it.
https://docs.python.org/3/howto/regex.html#using-regular-expressions

General comments on the regex:
- Why do yo match a single space *or* `.`?
- What happens with [the prefix-free parameters](https://warpx.readthedocs.io/en/latest/usage/parameters.html#overall-simulation-parameters) `authors`, `stop_time` and `max_step`?
- does this handle multi-line AMReX parser strings well? They are allowed in the C++ inputs and technically a user can also express multi-line strings in arguments of the Python API
- I think you look for `re.search` and not the extra cost of `split`:
```
re.search(pattern, string, flags=0)

    Scan through string looking for the first location where the regular expression pattern produces a match, and return a corresponding [match object](https://docs.python.org/3/library/re.html#match-objects). Return None if no position in the string matches the pattern; note that this is different from finding a zero-length match at some point in the string.
```
- I find the overall patch a bit verbose for what it does. I would maybe it a function call, or at least add a newline before the final `ff.write(f'{arg}\n')`. I would maybe even add it as a transformation to the `argv` list instead of patching the print loop.

Reply:
I can respond. I agree that it is standard practice to precompile an expression that is used repeatedly, but "slow" is relative. Doing some timings of the `re.split` command shows that for the few dozen lines of output here, it will take ~30 microseconds. Using `re.compile` does reduce it to ~10 microseconds.

- It is checking for space or `.` to handle both cases, with or without a prefix.
- The interface is already concatenating multiline input into a single line
- Yes, `re.search` would work as well, but the time difference would be negligible
- It could be compressed some by combing the two `if` statements (and I think `prefix_old = None` is better)
```
if prefix_old is not None and prefix_new != prefix_old:
    ff.write('\n')
```

Reply:
- slow: yes, just a note really
- `It is checking for space or . to handle both cases, with or without a prefix.` - I don't get why no-prefix starts with a space, do you introduce that intentionally for the Python wrapper?
- multiline: cool, ok

rest: as I said, reads a bit verbose to me, but otherwise ok.

Reply:
Thanks for merging. For the case without a prefix, the text looks like this `max_step = 100`. It is looking for that space before the `=`.